src/databricks/labs/ucx/framework/crawlers.py

```diff
@@ -119,26 +119,36 @@ def _snapshot(self, fetcher: ResultFn, loader: ResultFn) -> list[Result]:
         Args:
             fetcher: A function to fetch existing data.
             loader: A function to load new data.
+            force_refresh: Whether existing data should be ignored and forcibly replaced with data from the loader.
 
         Exceptions:
-        - If a runtime error occurs during fetching (other than "TABLE_OR_VIEW_NOT_FOUND"), the original error is
-          re-raised.
+        - Any errors raised by the fetcher are passed through, except for NotFound (usually due to
+          TABLE_OR_VIEW_NOT_FOUND) which is suppressed. All errors raised by the loader are passed through.
 
         Returns:
-        list[Result]: A list of data records, either fetched or loaded.
+            list[Result]: A list of data records, either fetched or loaded.
         """
-        logger.debug(f"[{self.full_name}] fetching {self._table} inventory")
-        try:
-            cached_results = list(fetcher())
-            if len(cached_results) > 0:
-                return cached_results
-        except NotFound:
-            pass
+        if force_refresh:
+            logger.debug(f"[{self.full_name}] ignoring any existing {self._table} inventory; refresh is forced.")
+        else:
+            logger.debug(f"[{self.full_name}] fetching {self._table} inventory")
+            try:
+                cached_results = list(fetcher())
+                if len(cached_results) > 0:
+                    return cached_results
+            except NotFound:
+                pass
         logger.debug(f"[{self.full_name}] crawling new batch for {self._table}")
         loaded_records = list(loader())
-        self._append_records(loaded_records)
+        self._update_snapshot(loaded_records, mode="overwrite")
         return loaded_records
 
-    def _append_records(self, items: Sequence[Result]):
+    def _update_snapshot(self, items: Sequence[Result], mode: Literal["append", "overwrite"] = "append") -> None:
         logger.debug(f"[{self.full_name}] found {len(items)} new records for {self._table}")
-        self._backend.save_table(self.full_name, items, self._klass, mode="append")
+        self._backend.save_table(self.full_name, items, self._klass, mode=mode)
+
+    def _append_records(self, items: Sequence[Result]):
+        self._update_snapshot(items, mode="append")
+
```

Comment:
PR marked as draft because although I thought we had some crawlers than incrementally append records, we don't seem to. I'm currently checking this and can probably remove this method once I've verified this.

Reply:
All crawlers now use the common flow, so the append/overwrite methods have been removed.